zerver/actions/message_flags.py

```diff
@@ -283,12 +284,24 @@ def do_update_message_flags(
             # currently subscribed to.
             subscribed_recipient_ids = get_subscribed_stream_recipient_ids_for_user(user_profile)
 
-            message_ids_in_unsubscribed_streams = set(
+            messages_in_unsubscribed_streams = set(
                 # Uses index: zerver_message_pkey
                 Message.objects.select_related("recipient")
                 .filter(id__in=messages, recipient__type=Recipient.STREAM)
                 .exclude(recipient_id__in=subscribed_recipient_ids)
-                .values_list("id", flat=True)
+                .values_list("id", "recipient__type_id")
+            )
+
+            message_ids_in_unsubscribed_streams = {
+                message[0] for message in messages_in_unsubscribed_streams
+            }
+
+            skipped_marking_unread_stream_ids = list(
+                {
+                    message[1]
+                    for message in messages_in_unsubscribed_streams
+                    if message[0] in messages
```

Comment:
wouldn't this condition always satisfy ?
as `messages_in_unsubscribed_streams` is from the query which filters the Messages table for the same ids.

Reply:
Yeah you are right updated it.